### PR TITLE
fix(AttemptCompletionTool): child tasks return to parent when parent status is active

### LINKED ISSUE #457: Subtask never returns to parent after a per-mode model/profile switch (attempt_completion silently waits for human)
**Zoo Code version:** 3.57.100116

**API provider/model:** Reproduces with per-mode profiles where the parent (orchestrator) and the spawned subtask use *different* model profiles (any providers).

**Related:** Part of #357 (Epic 2: Task Lifecycle Fixes). This is a **distinct** defect from the step 4/5 atomicity race — it is a return-**guard** that rejects the `active` parent state outright, so the parent never reaches those writes on the cross-profile hop.

### What happens

When an orchestrator/parent task delegates to a subtask whose mode is bound to a **different** API-configuration profile than the parent, the subtask completes its work and calls `attempt_completion`, but control **never returns to the parent**. The run silently waits for a manual click instead of auto-resuming the parent. Delegating to subtasks that share the parent's profile works fine; the first delegation to a *different*-profile mode is what breaks. It is order-independent — whichever profile is used first works, and the first switch to a different one breaks the next return.

### Steps to reproduce

1. Enable per-mode API configs (`modeApiConfigs`) so two modes use two different model profiles — e.g. an orchestrator on profile A and a sub-mode (e.g. `verifier`) on profile B.
2. From the orchestrator, delegate to a **same-profile** subtask. It returns to the parent correctly.
3. From the orchestrator, delegate (`new_task`) to a subtask on the **different** profile.
4. Let that subtask finish and call `attempt_completion`.

### Expected

Control returns to the parent task automatically (the parent resumes with the subtask result), the same as for same-profile subtasks.

### Actual

The subtask emits its completion result, but the parent is never re-opened; the flow stalls waiting for manual input.

### Root cause (verified)

The per-mode profile switch runs `handleModeSwitch` then `activateProviderProfile`, which leaves the parent task record as `status: "active"` while `awaitingChildId` is **still set to the child**. Two separate return-guards then reject because they require the parent status to be exactly `"delegated"`:

1. In the `attempt_completion` tool handler — the pre-check that decides whether to call `delegateToParent` requires `parent.status === "delegated"`. When it is `"active"`, it skips delegation and falls through to the "wait for user" path.
2. In `ClineProvider.reopenParentFromDelegation()` — the guard rejects unless `status === "delegated"`, logging `[reopenParentFromDelegation] Aborting: ... status=active`.

Because `awaitingChildId === childTaskId` already proves the parent is genuinely waiting for this exact child, both guards can safely also accept `status === "active"`.

### Suggested fix

Relax both guards to accept `"active"` in addition to `"delegated"`, only while `awaitingChildId` still points at the returning child:

- `attempt_completion` pre-check: `(parent.status === "delegated" || parent.status === "active") && parent.awaitingChildId === child`
- `reopenParentFromDelegation` guard: abort only when `status !== "delegated" && status !== "active"`

The unrelated cleanup paths (stack removal on `removeClineFromStack`, and `cancelTask`) should keep the strict `"delegated"`-only check.

### Workaround

Enabling **Lock the API configuration across all modes** (`lockApiConfigAcrossModes`) avoids the profile switch, so returns work — but every mode then runs on the same model, losing per-mode model routing.

**Roadmap alignment:** Reliability First — fixes a silent stall that blocks reliable multi-mode/agent workflows.

### PR DESCRIPTION
**Pull request by edelauna**
### Related GitHub Issue

Fixes: #457

### Description

Child tasks were failing to return to their parent after calling `attempt_completion` in v3.56+. The regression was reported on Discord; downgrading to v3.55 restored the behaviour.

**Root cause confirmed:** Issue #457 describes the exact sequence. When a same-profile child completes and the parent resumes, the parent status is transiently `"active"` while it awaits the next (different-profile) child. The guard in `AttemptCompletionTool` previously required `parentStatus === "delegated"` to proceed. Since the status was `"active"`, the guard rejected the return and the child fell through to standalone completion.

**Fix:** Both delegation guards now accept `"active"` in addition to `"delegated"` when `awaitingChildId` matches the returning child. Cancellation remains protected because cancellation clears `awaitingChildId`.

**Changes:**

- `AttemptCompletionTool`: Accept `parentStatus === "active"` when `awaitingChildId` matches the returning child.
- `ClineProvider#reopenParentFromDelegation`: Same guard relaxation for symmetry.
- `subtasks.test.ts`: E2e regression test reproducing the exact #457 sequence — same-profile child returns first, parent delegates to a different-profile child, both complete and parent resumes.
- `attemptCompletionTool.spec.ts`: Unit tests covering all guard boundary cases including the new path and the mismatched-`awaitingChildId` rejection.

### Test Procedure

- Unit tests: 33 passing (completion lifecycle suite)
- Full extension test suite: 6040 passing
- E2E subtask suite: 5 passing
- Type checks: clean

### Pre-Submission Checklist

- [x] **Issue Linked**: This PR is linked to an approved GitHub Issue (see "Related GitHub Issue" above).
- [x] **Scope**: My changes are focused on the linked issue (one major feature/fix per PR).
- [x] **Self-Review**: I have performed a thorough self-review of my code.
- [x] **Testing**: New and/or updated tests have been added to cover my changes (if applicable).
- [x] **Documentation Impact**: I have considered if my changes require documentation updates (see "Documentation Updates" section below).
- [x] **Contribution Guidelines**: I have read and agree to the [Contributor Guidelines](/CONTRIBUTING.md).

### Documentation Updates

- [x] No documentation updates are required.

### Additional Notes

The Discord reporter's original trigger has not been reproduced in isolation — history navigation while a child is active produces the same symptom but via a separate path. This PR fixes the confirmed #457 sequence. Diagnostic logging from the first commit remains in place to surface any remaining unknown trigger.

### Get in Touch

Discord: edelauna

## Summary by CodeRabbit

* **Tests**
  * Expanded end-to-end coverage with fast-completing child and cross-profile delegation scenarios; refactored tests for deterministic task sequencing and stronger cleanup.

* **Improvements**
  * Resuming parent tasks now recognizes additional parent states when children complete.
  * More informative diagnostics: enhanced log messages (including caller stack snippets) and cleaner fallback behavior when delegation is skipped.

### CODE REVIEW

File: apps/vscode-e2e/src/fixtures/subtasks.ts [hunk 1]
```diff
@@ -40,6 +54,54 @@ const completionAfterAnswer = (followupId: string, completionId: string) => ({
...
+					name: "new_task",
+					arguments: JSON.stringify({
+						mode: "ask",
+						message: SUBTASK_FAST_CHILD_PROMPT,
+					}),
+					id: "call_subtasks_fast_parent_new_task_001",
+				},
+			],
+		},
+	})
```
[the remark this answers is not in the document]
Comment (edelauna) [reply apps/vscode-e2e/src/fixtures/subtasks.ts:74]: Added in: https://github.com/Zoo-Code-Org/Zoo-Code/pull/510/commits/4610768a34f71fbbfe47efab6570379e174e8d0f

**Review by taltas — APPROVED** [review]
Nice PR

**Review by navedmerchant — APPROVED** [review]
lgtm
